### PR TITLE
Change isImage function to link instead of URL

### LINKED ISSUE #474: Bug: Discord: "Open URLs within Ferdium" behaves incorrectly
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.0.0-nightly.98

### What Operating System are you using?

macOS

### Operating System Version

12.4

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Ferdium version

_No response_

### Expected Behavior

When clicking an internal Discord link (eg: https://discord.com/channels/592103645835821068/592106336838352923/995865013057245205), it should not reload the Discord Ferdium tab, but rather instantly jump to the conversation to which the URL leads.

This worked in the past, but now seems to be broken. This causes very difficult behavior when moving between conversations that other users link to.

### Actual Behavior

Currently when you click an internal Discord link within the Ferdium tab/service, it reloads Discord and then jumps to the location in question. 

### Steps to reproduce

1. Modify service settings and turn on "Open URLs within Ferdium"
2. Find a chat URL (example provided above)
3. Click it
4. Notice how it reloads the entire service/tab.

If you disable "Open URLs within Ferdium" it now simply does not go anywhere or goes to the default browser for the OS.

### Debug link

_No response_

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SpecialAro**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Change isImage function to link instead of URL

#### Motivation and Context
This change is needed to improve how `isImage` function works. This PR needs to be merge **BEFORE** or together with https://github.com/ferdium/ferdium-recipes/pull/120.

This fixes https://github.com/ferdium/ferdium-app/issues/474

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes

Change isImage function to link instead of URL

### CODE REVIEW

File: src/webview/lib/RecipeWebview.ts [hunk 1]
```diff
@@ -93,12 +93,14 @@ class RecipeWebview {
...
-   * Find if url contains image
+   * Find if link contains image
    *
    * @param  {string | number | undefined | null} text to be parsed
    */
-  isImage(url): boolean {
-    return /\.(jpg|jpeg|png|webp|avif|gif|svg)$/.test(url.split(/[#?]/)[0]);
+  isImage(link): boolean {
+    const roles = ['img'];
+    const { role } = link.dataset;
```
Comment (vraravam) [inline src/webview/lib/RecipeWebview.ts:107]: What if `link` is undefined? also, what if the link does not have a dataset? For eg, 
`<a href='https://www.google.com'>Search in google</a>` is a link without a dataset.

**Review by vraravam — CHANGES_REQUESTED** [review]

File: src/webview/lib/RecipeWebview.ts (hunk 1 above)
Comment (SpecialAro) [reply src/webview/lib/RecipeWebview.ts:107]: > For eg, `<a href='https://www.google.com'>Search in google</a>` is a link without a dataset.

Tested this an, in fact, it has a dataset but we should check if dataset.role is undefined first, in fact.

**Review by Gaboris — APPROVED** [review]
I can't notice any bugs on paper so we'll see if it has any conflicts in action. 👍

**Review by vraravam — APPROVED** [review]

File: src/webview/lib/RecipeWebview.ts [hunk 2]
```diff
@@ -93,12 +93,27 @@ class RecipeWebview {
...
+
+    if (typeof role !== 'undefined') {
+      const roles = ['img'];
+      return roles.includes(role);
+    }
+
+    const url = link.getAttribute('href');
+
+    return /\.(jpg|jpeg|png|webp|avif|gif|svg)($|\?|:)/.test(
+      url.split(/[#?]/)[0],
```
Comment (cino) [inline src/webview/lib/RecipeWebview.ts]: For readability in this case, I would extract the regex to a const above

**Review by vraravam — APPROVED** [review]
LGTM!

**Pull request merged**